core/src/main/scala/chisel3/Printf.scala

```diff
@@ -109,19 +78,7 @@ object printf {
     * @param data format string varargs containing data to print
     */
   def apply(fmt: String, data: Bits*)(implicit sourceInfo: SourceInfo): Printf =
-    macro _applyMacroWithInterpolatorCheck
-
-  def _applyMacroWithInterpolatorCheck(
-    c:          blackbox.Context
-  )(fmt:        c.Tree,
-    data:       c.Tree*
-  )(sourceInfo: c.Tree
-  ): c.Tree = {
-    import c.universe._
-    _checkFormatString(c)(fmt)
-    val apply_impl_do = symbolOf[this.type].asClass.module.info.member(TermName("printfWithReset"))
-    q"$apply_impl_do(_root_.chisel3.Printable.pack($fmt, ..$data))($sourceInfo)"
-  }
+    macro PrintfMacrosCompat._applyMacroWithInterpolatorCheck
```

Comment:
FWIW, I'm not sure that this will work, I'm pretty sure `macro` the keyword only works in Scala 2. I think we have to do a similar "veneer" thing where the full public method exists only in `src/main/scala-2`:
```scala
def apply(fmt: String, data: Bits*)(implicit sourceInfo: SourceInfo): Printf =
  macro PrintfMacrosCompat._applyMacroWithInterpolatorCheck
```

Reply:
this in fact works in scala3 since the scala3 compiler can read scala2 macro definitions. the compiler can pick between scala2 and scala3 definitions, so while transitioning we need to add scala3 macro definitions while keeping the scala2 ones in the same artifact

Reply:
Relevant documentation: https://docs.scala-lang.org/scala3/guides/migration/tutorial-macro-mixing.html